modules/raft/src/main/java/org/apache/ignite/internal/raft/server/impl/JraftServerImpl.java

```diff
@@ -856,4 +860,54 @@ public void onLeaderStart(long term) {
             listener.onLeaderStart();
         }
     }
+
+    /** Returns raft group state accessor. */
+    public <G extends ReplicationGroupId, L> RaftGroupStateProvider<G, L> getGroupStateProvider(
```

Comment:
I'm not sure that fully get the idea of the provided code. I'd rather implement it in a bit different manner, precisely by adjusting UpdateMinimumActiveTxBeginTimeCommand handling in a way not only to update minActiveTxBeginTime but also to return last flushed one. WDYT?

Reply:
@sanpwc we need to collect timestamps from ALL replication groups (primary and non-primary), can you clarify is this possible to do with proposed approach?
Can you briefly describe how to do this?

Currently we send a request to PartitionReplicaListener and apply the raft command, and as far as I remember we don't even know when the command will be applied to ALL replicas in the group (we only know when it will be applied to the primary replica) :thinking: .

Reply:
> we need to collect timestamps from ALL replication groups (primary and non-primary), can you clarify is this possible to do with proposed approach?

Unfortunately no. And I've missed that, so please ignore my comment.

Can you please show a place where you evaluate lastSnapshotted(flushed) tx start time?